### PR TITLE
Add n_cell property to NStepRNN family

### PR DESCRIPTION
**Pull request by levelfour**
ChainerMN needs this change in order to parallelize NStepRNN ([MultiNodeNStepRNN](https://github.com/chainer/chainermn/blob/master/chainermn/links/n_step_rnn.py)), where each cell is transferred to the neighbour process and thus it needs the number of cells in advance.

### Can it be completed inside of ChainerMN?

Unfortunately, no. Currently, we prepare [a table to obtain the number of cells](https://github.com/chainer/chainermn/pull/205/files#diff-1ed392351c4f47516bae8214681548eeR11), but this solution requires high maintenance cost to keep consistency between Chainer and ChainerMN, and hinders users to define original RNNs.

I tried whether we can avoid hard-coding, but I couldn't. If there is any nice solution, I would be happy to update.

### CODE REVIEW

File: chainer/links/connection/n_step_rnn.py [hunk 1]
```diff
@@ -116,6 +116,14 @@ def rnn(self, *args):
         """
         raise NotImplementedError
 
+    @property
+    def n_cell(self):
```
Comment (unnonouno) [inline chainer/links/connection/n_step_rnn.py]: `n_cells` is better.

File: chainer/links/connection/n_step_rnn.py [hunk 2]
```diff
@@ -116,6 +116,14 @@ def rnn(self, *args):
...
+        """Return the number of cells.
```
Comment (unnonouno) [inline chainer/links/connection/n_step_rnn.py]: Returns

**Comment by unnonouno** [comment]
Can you write tests please?

**Comment by levelfour** [comment]
What kind of tests are expected in this case? Is it appropriate to just check return values of `n_cells`?

**Comment by unnonouno** [comment]
Yes, it is enough to check the return value.

**Comment by levelfour** [comment]
I added tests. Could you take a glance again?

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**